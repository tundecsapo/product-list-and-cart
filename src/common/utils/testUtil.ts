export const baseProduct = {
  id: "62863b689c1bcb9946a0c8ac",
  name: "Bolts",
  img: "https://media.gettyimages.com/photos/bolts-and-nuts-picture-id175425827?s=2048x2048",
  availableAmount: 1000,
  minOrderAmount: 20,
  price: 1,
  selectedAmount: 20,
};
