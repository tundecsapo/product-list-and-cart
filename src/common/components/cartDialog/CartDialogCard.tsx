import Box from "@mui/material/Box";
import Card from "@mui/material/Card";
import CardContent from "@mui/material/CardContent";
import CardMedia from "@mui/material/CardMedia";
import Typography from "@mui/material/Typography";
import { SelectedProduct } from "../../types";
import { priceFormatter } from "../../utils/priceFormatter";

export const CartDialogCard = ({ item }: { item: SelectedProduct }) => {
  return (
    <Card
      sx={{
        display: "flex",
        boxShadow: "none",
        borderBottom: "1px solid rgba(0,0,0,0.2)",
      }}
    >
      <CardMedia
        component="img"
        sx={{ width: 100, maxHeight: 92 }}
        image={item.img}
        alt={`Image of ${item.name}`}
      />
      <Box sx={{ display: "flex", flexDirection: "column" }}>
        <CardContent sx={{ flex: "1 0 auto" }}>
          <Typography component="div" variant="body1">
            {item.name}
          </Typography>
          <Typography
            variant="subtitle1"
            color="text.secondary"
            component="div"
          >
            {`${item.selectedAmount} x ${priceFormatter.format(item.price)}`}
          </Typography>
        </CardContent>
      </Box>
    </Card>
  );
};
