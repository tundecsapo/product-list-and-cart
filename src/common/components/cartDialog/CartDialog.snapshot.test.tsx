import renderer from "react-test-renderer";
import { baseProduct } from "../../utils/testUtil";
import { CartDialogCard } from "./CartDialogCard";

it("renders correctly", () => {
  const tree = renderer.create(<CartDialogCard item={baseProduct} />).toJSON();
  expect(tree).toMatchSnapshot();
});
