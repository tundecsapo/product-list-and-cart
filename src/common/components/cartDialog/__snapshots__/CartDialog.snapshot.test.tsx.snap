// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-16wdysy-MuiCard-root{overflow:hidden;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;box-shadow:none;border-bottom:1px solid rgba(0,0,0,0.2);}",
      }
    }
    data-emotion="css 16wdysy-MuiCard-root"
  />,
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-m5pjwk-MuiPaper-root-MuiCard-root{background-color:#fff;color:rgba(0, 0, 0, 0.87);-webkit-transition:box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;border-radius:4px;box-shadow:0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);overflow:hidden;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;box-shadow:none;border-bottom:1px solid rgba(0,0,0,0.2);}",
      }
    }
    data-emotion="css m5pjwk-MuiPaper-root-MuiCard-root"
  />,
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-m5pjwk-MuiPaper-root-MuiCard-root"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1lpirw8-MuiCardMedia-root{display:block;-webkit-background-size:cover;background-size:cover;background-repeat:no-repeat;-webkit-background-position:center;background-position:center;width:100%;object-fit:cover;width:100px;max-height:92px;}",
        }
      }
      data-emotion="css 1lpirw8-MuiCardMedia-root"
    />
    <img
      alt="Image of Bolts"
      className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-1lpirw8-MuiCardMedia-root"
      src="https://media.gettyimages.com/photos/bolts-and-nuts-picture-id175425827?s=2048x2048"
    />
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-j7qwjs{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;}",
        }
      }
      data-emotion="css j7qwjs"
    />
    <div
      className="MuiBox-root css-j7qwjs"
    >
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-1qaznj3-MuiCardContent-root{padding:16px;-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;}.css-1qaznj3-MuiCardContent-root:last-child{padding-bottom:24px;}",
          }
        }
        data-emotion="css 1qaznj3-MuiCardContent-root"
      />
      <div
        className="MuiCardContent-root css-1qaznj3-MuiCardContent-root"
      >
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-ahj2mt-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;}",
            }
          }
          data-emotion="css ahj2mt-MuiTypography-root"
        />
        <div
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Bolts
        </div>
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-ecpxie-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.75;letter-spacing:0.00938em;color:rgba(0, 0, 0, 0.6);}",
            }
          }
          data-emotion="css ecpxie-MuiTypography-root"
        />
        <div
          className="MuiTypography-root MuiTypography-subtitle1 css-ecpxie-MuiTypography-root"
        >
          20 x $1
        </div>
      </div>
    </div>
  </div>,
]
`;
